src/gg.regression.unity.bots/Editor/Scripts/RGOverlay/IRGBotFinder.cs

```diff
@@ -0,0 +1,62 @@
+using UnityEditor;
+using UnityEditor.Callbacks;
+using UnityEngine;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace RegressionGames
+{
+    public class IRGBotFinder : UnityEditor.Editor
+    {
+        [DidReloadScripts]
+        private static void OnScriptsReloaded()
+        {
+            // Only run this method in the Unity editor, not at runtime
+            if (!EditorApplication.isPlayingOrWillChangePlaymode || EditorApplication.isPlaying)
+            {
+                return;
+            }
+
+            // Find the IRGBotList ScriptableObject in your project
+            IRGBotList botList = LoadBotList();
+
+            if (botList == null)
+            {
+                Debug.LogWarning("IRGBotList ScriptableObject not found");
+                return;
+            }
+
+            // Use TypeCache to find all types that implement IRGBot
+            var botTypes = TypeCache.GetTypesDerivedFrom<IRGBot>()
+                .Where(t => typeof(MonoBehaviour).IsAssignableFrom(t) && !t.IsAbstract);
+
+            List<IRGBotEntry> entries = new List<IRGBotEntry>();
+
+            foreach (var type in botTypes)
+            {
+                var qualifiedName = type.AssemblyQualifiedName;
+                if (!string.IsNullOrEmpty(qualifiedName))
+                {
+                    entries.Add(new IRGBotEntry
+                    {
+                        botName = type.Name,
+                        qualifiedName = qualifiedName,
+                    });
+                }
+            }
+
+            botList.botEntries = entries.ToArray();
+
+            // Mark the ScriptableObject as dirty to ensure it saves
+            EditorUtility.SetDirty(botList);
+            AssetDatabase.SaveAssets();
+            AssetDatabase.Refresh();
+        }
+
+        private static IRGBotList LoadBotList()
+        {
+            string packagePath = "Packages/gg.regression.unity.bots/Runtime/Resources/RGBotList.asset";
```

Comment:
I thought the discussion yesterday concluded with us copying this default into their project and editing the one in their project ?

Reply:
After the package path was fixed, it's working as expected with the scriptable object in the package. It should never have to be seen or updated by a developer so if there's no technical reason to copy it over, I think we should leave it. What do you think? @RG-nAmKcAz @vontell

Reply:
Quite simply, you can't modify that file in our package unless you're an RG developer loading it from your local filesystem.

When using the package from a tarball or from git that file isn't writeable.

![image](https://github.com/Regression-Games/RGUnityBots/assets/112959031/818b01df-e438-44ea-afff-48cc818d3f32)

Reply:
Implemented 👍